examples/elasticsearch_outlet.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import random
+import time
+from typing import List
+
+import elasticsearch
+from databay import Inlet, Link, Outlet
+from databay.planners import ApsPlanner
+from databay.record import Record
+
+TEXT = """
+Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+Phasellus ex erat, viverra tincidunt tempus eget, hendrerit sed ligula. 
+Quisque mollis nibh in imperdiet porttitor. Nulla bibendum lacus et est lobortis porta.
+Nulla sed ligula at odio volutpat consectetur. Sed quis augue ac magna porta imperdiet interdum eu velit. 
+Integer pretium ultrices urna, id viverra mauris ultrices ut. Etiam aliquet tellus porta nisl eleifend, non hendrerit nisl sodales. 
+Aliquam eget porttitor enim. 
+"""
+
+_LOGGER = logging.getLogger('databay.elasticsearch_outlet')
+
+
+class ElasticSearchIndexerOutlet(Outlet):
+    " An example outlet for indexing text documents from any `Inlet`."
+
+    def __init__(self,
+                 es_client: elasticsearch.Elasticsearch,
+                 index_name: str,
+                 overwrite_documents: bool = True):
+        super().__init__()
+        self.es_client = es_client
+        self.index_name = index_name
+
+        # if true existing documents will be overwritten
+        # otherwise we will skip indexing and log that document id exists in index.
+        self.overwrite_documents = overwrite_documents
+
+        if self.es_client.indices.exists(self.index_name):
+            pass
+        else:
+            raise RuntimeError(f"Index '{self.index_name}' does not exist ")
```

Comment:
This `if` seems a bit upside-down, doesn't it? Why not flip the boolean with a `if not self.es_client...` and avoid the `pass` block altogether?